modules/swagger-codegen/src/main/resources/swift3/Models.mustache

```diff
@@ -202,6 +205,20 @@ class Decoders {
             }{{/isEnum}}
             {{^isEnum}}instance.{{name}} = Decoders.decodeOptional(clazz: {{{baseType}}}.self, source: sourceDictionary["{{baseName}}"] as AnyObject?){{/isEnum}}{{/allVars}}
             {{/unwrapRequired}}
+
+            {{#parent}}
+            for key in keys {
+                propsDictionary.removeValue(forKey: key)
+            }
+
+            for key in propsDictionary.keys {
+                if let decodedValue = Decoders.decodeOptional(clazz: String.self, source: propsDictionary[key] as AnyObject?) {
+                    instance[key as! String] = decodedValue
+                } else {
+                    instance[key as! String] = nil
```

Comment:
Actually yes good point.  Im just headed out, ill update and run a test when I get back.